src/data_source/fetching.rs

```diff
@@ -1401,8 +1402,16 @@ where
                 );
                 loop {
                     let res = async {
+                        let start = chunk[0].height;
+                        let aggregate = if start == 0 {
+                            Aggregate::default()
+                        } else {
+                            let mut tx = self.read().await.context("opening transaction")?;
+                            tx.aggregate((start - 1).try_into()?).await?
+                        };
+
                         let mut tx = self.write().await.context("opening transaction")?;
-                        tx.update_aggregates(&chunk).await?;
+                        tx.update_aggregates(aggregate, &chunk).await?;
```

Comment:
Since we're now passing in the previous aggregates to each iteration, we have the opportunity to do something kinda cool here...we could have `update_aggregates` also return the updated aggregates, and then we can just keep `aggregate` in a local variable here, updating each iteration of the loop. We would only need to read it from the database once before the first iteration.

This would actually save one database interaction every iteration of the loop which is pretty good. Similar to how the state update loop works in the confirmation layer